staging/src/k8s.io/apiserver/pkg/apis/audit/validation/validation.go

```diff
@@ -60,3 +65,40 @@ func validateLevel(level audit.Level, fldPath *field.Path) field.ErrorList {
 		return field.ErrorList{field.NotSupported(fldPath, level, validLevels)}
 	}
 }
+
+func validateNonResourceURLs(urls []string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	for i, url := range urls {
+		if url == "*" {
+			continue
+		}
+
+		if !strings.HasPrefix(url, "/") {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(i), url, "non-resource URL rules must begin with a '/' character"))
+		}
+
+		if strings.ContainsRune(url[:len(url)-1], '*') {
```

Comment:
An empty url matches nothing:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/audit/policy/checker.go#L112
I think an empty url doesn't make any sense, so empty url is invalid for me.